src/plugins/controls/public/control_types/range_slider/range_slider_embeddable.tsx

```diff
@@ -0,0 +1,312 @@
+  ReduxEmbeddableWrapperPropsWithChildren<RangeSliderEmbeddableInput>
+>(LazyReduxEmbeddableWrapper);
+
+const diffDataFetchProps = (
+  current?: RangeSliderDataFetchProps,
+  last?: RangeSliderDataFetchProps
+) => {
+  if (!current || !last) return false;
+  const { filters: currentFilters, ...currentWithoutFilters } = current;
+  const { filters: lastFilters, ...lastWithoutFilters } = last;
+  if (!deepEqual(currentWithoutFilters, lastWithoutFilters)) return false;
+  if (!compareFilters(lastFilters ?? [], currentFilters ?? [], COMPARE_ALL_OPTIONS)) return false;
+  return true;
+};
+
+interface RangeSliderDataFetchProps {
+  fieldName: string;
+  dataViewId: string;
+  query?: ControlInput['query'];
+  filters?: ControlInput['filters'];
+}
+
+const fieldMissingError = (fieldName: string) =>
+  new Error(`field ${fieldName} not found in index pattern`);
+
+export class RangeSliderEmbeddable extends Embeddable<RangeSliderEmbeddableInput, ControlOutput> {
+  public readonly type = RANGE_SLIDER_CONTROL;
+  public deferEmbeddableLoad = true;
+
+  private subscriptions: Subscription = new Subscription();
+  private node?: HTMLElement;
+
+  // Controls services
+  private dataService: ControlsDataService;
+  private dataViewsService: ControlsDataViewsService;
+
+  // Internal data fetching state for this input control.
+  private dataView?: DataView;
+  private field?: DataViewField;
+
+  // State to be passed down to component
+  private componentState: RangeSliderComponentState;
+  private componentStateSubject$ = new BehaviorSubject<RangeSliderComponentState>({
+    min: '',
+    max: '',
+    loading: true,
+    fieldFormatter: (value: string) => value,
+  });
+
+  constructor(input: RangeSliderEmbeddableInput, output: ControlOutput, parent?: IContainer) {
+    super(input, output, parent); // get filters for initial output...
+
+    // Destructure controls services
+    ({ data: this.dataService, dataViews: this.dataViewsService } = pluginServices.getServices());
+
+    this.componentState = {
+      min: '',
+      max: '',
+      loading: true,
+      fieldFormatter: (value: string) => value,
+    };
+    this.updateComponentState(this.componentState);
+
+    this.initialize();
+  }
+
+  private initialize = async () => {
+    const initialValue = this.getInput().value;
+    if (initialValue) {
+      await this.fetchMinMax();
+    }
+    this.setInitializationFinished();
+    this.setupSubscriptions();
+  };
+
+  private setupSubscriptions = () => {
+    const dataFetchPipe = this.getInput$().pipe(
+      map((newInput) => ({
+        lastReloadRequestTime: newInput.lastReloadRequestTime,
+        dataViewId: newInput.dataViewId,
+        fieldName: newInput.fieldName,
+        timeRange: newInput.timeRange,
+        filters: newInput.filters,
+        query: newInput.query,
+      })),
+      distinctUntilChanged(diffDataFetchProps)
+    );
+
+    // fetch available min/max when input changes or when search string has changed
```

Comment:
super small nit: You don't have a search string in the range slider! :)

```suggestion
    // fetch available min/max when input changes
```